### ISSUE TITLE
Seeing error when trying to serialize an enum with rmp-serde

### ISSUE THREAD

**Issue by crhino** [3Hren/msgpack-rust#42]
I am trying to serialize an enum with `rmp-serde` and I am running into an error: 
```
`Err` value: DeserializeError(Syntax("syntax error: expected an enum"))`
```
which seems to come from serde's [default implementation](https://github.com/serde-rs/serde/blob/a1bd0c16675c37c7b81d5907dc059457b93971e3/serde/src/de/mod.rs#L388) of `visit_enum`.

I see that you have [tests surround serializing enums](https://github.com/3Hren/msgpack-rust/blob/60d872e2eed97e09bede6b48884285136341f5b2/rmp-serde/tests/deserializer.rs#L393), which is almost exactly what I am trying to do. I am wondering if there is something that I am missing here. Taking a look at your [Travis builds](https://travis-ci.org/3Hren/msgpack-rust/jobs/78282279) it does not seem like those tests are being run there.

I already have `serde_macros` enabled and have `custom_derive` and `plugin` features, not sure what else I need.

Any help would be much appreciated!

**Comment by crhino** [comment]
Figured out how to get the tests running, looks like they have the same failure.
```
$ cargo test --features "serde_macros"
   Compiling rmp-serde v0.7.0 (file:///home/chris/workspace/msgpack-rust/rmp-serde)
     Running target/debug/deserializer-496fb9d6ea316c12

running 37 tests
test fail_bool_from_fixint ... ok
test fail_null_from_reserved ... ok
test fail_option_u8_from_reserved ... ok
test fail_tuple_len_mismatch ... ok
test pass_bin16_into_bytebuf ... ok
test fail_u32_from_u64 ... ok
test fail_enum_sequence_mismatch ... FAILED
test pass_bin8_into_bytebuf ... ok
test fail_enum_overflow ... FAILED
test pass_bin32_into_bytebuf ... ok
test pass_bool ... ok
test pass_enum_variant_with_arg ... FAILED
test pass_enum ... FAILED
test pass_f32 ... ok
test pass_i16 ... ok
test pass_i64 ... ok
test pass_f64 ... ok
test pass_i8 ... ok
test pass_i32 ... ok
test pass_isize ... ok
test pass_option_some_null ... ok
test pass_option_some ... ok
test pass_null ... ok
test pass_struct ... ok
test pass_option_none ... ok
test pass_struct_map ... ok
test pass_string ... ok
test pass_tuple ... ok
test pass_struct_enum_with_arg ... FAILED
test pass_tuple_struct ... ok
test pass_map ... ok
test pass_u32 ... ok
test pass_u16 ... ok
test pass_u8 ... ok
test pass_u64 ... ok
test pass_usize ... ok
test pass_vector ... ok

failures:

---- fail_enum_sequence_mismatch stdout ----
	thread 'fail_enum_sequence_mismatch' panicked at 'unexpected result: Some(Syntax("syntax error: expected an enum"))', tests/deserializer.rs:429


---- fail_enum_overflow stdout ----
	thread 'fail_enum_overflow' panicked at 'unexpected result: Some(Syntax("syntax error: expected an enum"))', tests/deserializer.rs:449


---- pass_enum_variant_with_arg stdout ----
	thread 'pass_enum_variant_with_arg' panicked at 'called `Result::unwrap()` on an `Err` value: Syntax("syntax error: expected an enum")', ../src/libcore/result.rs:734


---- pass_enum stdout ----
	thread 'pass_enum' panicked at 'called `Result::unwrap()` on an `Err` value: Syntax("syntax error: expected an enum")', ../src/libcore/result.rs:734


---- pass_struct_enum_with_arg stdout ----
	thread 'pass_struct_enum_with_arg' panicked at 'called `Result::unwrap()` on an `Err` value: Syntax("syntax error: expected an enum")', ../src/libcore/result.rs:734



failures:
    fail_enum_overflow
    fail_enum_sequence_mismatch
    pass_enum
    pass_enum_variant_with_arg
    pass_struct_enum_with_arg

test result: FAILED. 32 passed; 5 failed; 0 ignored; 0 measured
```

**Comment by 3Hren** [comment]
Sorry for a long waiting. 

My fault - currently enum deserializing is broken. I'll try to fix today.

**Comment by 3Hren** [comment]
Please take a look for https://github.com/3Hren/msgpack-rust/pull/44. It should fix your issues.

**Comment by crhino** [comment]
@3Hren Looks good! Thanks.

**Comment by 3Hren** [comment]
Resolved with #44.
